config-plugins/withAndroidWhatsappPackage.js

```diff
@@ -0,0 +1,30 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { withAndroidManifest } = require('@expo/config-plugins');
+
+function addIntentToQuery({ mod, intent }) {
```

Comment:
why is this `addIntentToQuery` a bit different than in the other file? couldn't this be the same function exported in a file and used in both config-plugin files?

Reply:
I updated the code in this way. 

```
export const addIntentToQuery = ({ mod, intent }) => {
  const copyMod = { ...mod };
  if (!copyMod.modResults.manifest.queries) {
    copyMod.modResults.manifest.queries = [{ intent: [] }];
  } else if (!copyMod.modResults.manifest.queries[0]) {
    copyMod.modResults.manifest.queries = [];
  } else if (!copyMod.modResults.manifest.queries[0].intent?.[0]) {
    copyMod.modResults.manifest.queries[0].intent = [];
  }

  copyMod.modResults.manifest.queries.push({
    intent: { action: [{ $: { 'android:name': intent.actionName } }] },
    package: [{ $: { 'android:name': intent.actionName } }],
    ...(intent.dataScheme && { data: [{ $: { 'android:scheme': intent.dataScheme } }] })
  });

  return copyMod;
};
```

There is also a difference between Whatsapp and Mail plugins. I need to add Whatsapp to `queries` as a `package`. I need to add the Mail feature as an `action` in the `intent`. I didn't want to use it in one place because I thought extra controls might make the code more complex.

I edited `addIntentToQuery` and exported it. When I ran the `prebuild` command, I got the following error.

<img width="257" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/e794ad6a-d690-4a89-b105-8f1ef30fbd67">

---

After running the plugin, AndroidManifest needs to be updated as follows.
|before|after|
|--|--|
<img width="422" alt="image" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/bf261faa-5fd6-4f3a-b706-3529ee1285fa"> | <img width="423" alt="SCR-20230703-ksgb" src="https://github.com/smart-village-solutions/smart-village-app-app/assets/11755668/ef39aad8-363e-49ee-b437-a53c257306ee">

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/001f367f34ae03dade0fa83279b273fd37231cee